src/sentry/runner/commands/run.py

```diff
@@ -505,6 +508,27 @@ def occurrences_ingest_consumer(**options):
         run_processor_with_signals(consumer)
 
 
+def delay_kafka_rebalance(configured_delay: int) -> None:
+    """
+    get current time.
+    get the seconds part of the current time.
+
+    get the seconds delay to sleep from an option.
+
+    and then continue to sleep for 0.5 seconds until you get to the exact second when you want to start/stop the app.
+    """
+    now = float(datetime.now().strftime("%S.%f"))
+
+    next_tick, remainder = divmod(now, configured_delay)
+    if remainder > 0:
+        next_tick += 1
+
+    seconds_sleep = (configured_delay * next_tick) - now
+    while seconds_sleep >= 0.5:
```

Comment:
I wonder if this is enough "granularity", i.e. if different calls to `subscribe` happen ~0.5 seconds of each other, would they still be grouped as part of the same rebalance?